Octokit/Clients/RepositoriesClient.cs

```diff
@@ -163,6 +163,35 @@ public Task Delete(long repositoryId)
             return ApiConnection.Delete(ApiUrls.Repository(repositoryId));
         }
 
+        /// <summary>
+        /// Transfers the ownership of the specified repository.
+        /// </summary>
+        /// <remarks>
+        /// See the <a href="https://developer.github.com/v3/repos/#transfer-a-repository">API documentation</a> for more information.
+        /// </remarks>
+        /// <param name="currentOwner">The current owner of the repository</param>
+        /// <param name="name">The name of the repository</param>
+        /// <param name="repositoryTransfer">Repository transfer information</param>
+        /// <returns>A <see cref="Repository"/></returns>
+        public Task<Repository> Transfer(string currentOwner, string name, RepositoryTransfer repositoryTransfer)
+        {
+            return ApiConnection.Post<Repository>(ApiUrls.RepositoryTransfer(currentOwner, name), repositoryTransfer, AcceptHeaders.RepositoryTransferPreview);
```

Comment:
This is missing the `Ensure.ArgumentNotNullOrEmptyString()` checks for `currentOwner`, `name` and `Ensure.ArgumentNotNull()` check for `repositoryTransfer` (and the unit tests are not picking it up due to not being `await`'ed)